### PR TITLE
Added RemoveEmpty.scala, which removes Empty and nested Blocks

### PR DESCRIPTION
**Pull request by azidar**
A utility pass other Firrtl compilers can use. Also makes testing easier because you can remove Empty statements that may be introduced via your compiler pass (but you don't want to have them in the check string).

### CODE REVIEW

**Comment by jackkoenig** [comment]
I have a utility written for this in ExpandWhens. We should remove it from there and put some equivalent implementation in Utils that should then be used by this pass. I think mine is neat but whatever.

https://github.com/ucb-bar/firrtl/blob/master/src/main/scala/firrtl/passes/ExpandWhens.scala#L74

File: src/main/scala/firrtl/passes/RemoveEmpty.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          Block(newStmts.toSeq)
+        case s => s map onScope
+      }
+    }
+    m match {
+      case m: Module => Module(m.info, m.name, m.ports, onScope(m.body))
+      case m: ExtModule => m
+    }
+  }
+  def run(c: Circuit): Circuit = Circuit(c.info, c.modules.map(onModule _), c.main)
```
Comment (jackkoenig) [inline src/main/scala/firrtl/passes/RemoveEmpty.scala]: Prefer infix notation ie.
```scala
c.modules map onModule
```

**Comment by azidar** [comment]
Ah ok, I'll move it to utils instead.
